src/EA.Weee.RequestHandlers.Tests.DataAccess/Admin/GetProducerDetails/GetProducerComplianceYearDataAccessTests.cs

```diff
@@ -0,0 +1,192 @@
+﻿namespace EA.Weee.RequestHandlers.Tests.DataAccess.Admin.GetProducerDetails
+{
+    using RequestHandlers.Admin.GetProducerDetails;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Weee.Tests.Core.Model;
+    using Xunit;
+
+    public class GetProducerComplianceYearDataAccessTests
+    {
+        [Fact]
+        public async Task GetComplianceYears_ReturnsComplianceYearForRegisteredProducer()
+        {
+            using (DatabaseWrapper database = new DatabaseWrapper())
+            {
+                // Arrange
+                ModelHelper helper = new ModelHelper(database.Model);
+
+                var scheme = helper.CreateScheme();
+                var memberUpload = helper.CreateSubmittedMemberUpload(scheme);
+                memberUpload.ComplianceYear = 2016;
+
+                var producer1 = helper.CreateProducerAsCompany(memberUpload, "AAA");
+
+                database.Model.SaveChanges();
+
+                var dataAccess = new GetProducerComplianceYearDataAccess(database.WeeeContext);
+
+                // Act
+                var result = await dataAccess.GetComplianceYears("AAA");
+
+                // Assert
+                Assert.Single(result);
+                Assert.Contains(2016, result);
+            }
+        }
+
+        [Fact]
+        public async Task GetComplianceYears_ReturnsComplianceYearForSpecifiedProducerRegistrationNumberOnly()
```

Comment:
As discussed, this is as expected and will return all registered producers, regardless whether they have any data returns.